core/src/identity/secp256k1.rs

```diff
@@ -74,28 +68,30 @@ impl From<Keypair> for SecretKey {
 
 /// A Secp256k1 secret key.
 #[derive(Clone)]
-pub struct SecretKey(secp::key::SecretKey);
-
-/// View the bytes of the secret key.
-impl AsRef<[u8]> for SecretKey {
-    fn as_ref(&self) -> &[u8] {
-        &self.0[..]
-    }
-}
+pub struct SecretKey(secp256k1::SecretKey);
 
 impl SecretKey {
     /// Generate a new Secp256k1 secret key.
     pub fn generate() -> SecretKey {
-        SecretKey(secp::key::SecretKey::new(&mut secp::rand::thread_rng()))
+        let mut r = rand::thread_rng();
+        let mut b = [0; secp256k1::util::SECRET_KEY_SIZE];
+        // This is how it is done in `secp256k1::SecretKey::random` which
+        // we do not use here because it uses `rand::Rng` from rand-0.4.
+        loop {
+            r.fill_bytes(&mut b);
+            if let Ok(k) = secp256k1::SecretKey::parse(&b) {
+                return SecretKey(k)
+            }
+        }
     }
 
     /// Create a secret key from a byte slice, zeroing the slice on success.
     /// If the bytes do not constitute a valid Secp256k1 secret key, an
     /// error is returned.
     pub fn from_bytes(mut sk: impl AsMut<[u8]>) -> Result<SecretKey, DecodingError> {
         let sk_bytes = sk.as_mut();
-        let secret = secp::key::SecretKey::from_slice(&*sk_bytes)
-            .map_err(|e| DecodingError::new("Secp256k1 secret key", e))?;
+        let secret = secp256k1::SecretKey::parse_slice(&*sk_bytes)
+            .map_err(|_| DecodingError::new("failed to parse secp256k1 secret key"))?;
```

Comment:
I just realized that there is a missing `sk_bytes.zeroize();` below this line, i.e. the function currently does not do what the documentation says. Since this is an oversight unrelated to this PR, I created a separate, tiny PR for that: https://github.com/libp2p/rust-libp2p/pull/1040.